### DISCUSSION TITLE
using-kotlin-scope-function-correctly/

### DISCUSSION THREAD

**Discussion by giscus** [Hyeon9mak/Hyeon9mak.github.io#163] [Blog comments]
# using-kotlin-scope-function-correctly/

let, also 와 같은 Scope function 을 오래전부터 사용해왔지만 정확히 어떤 상황에 어떤 Scope function 을 사용하면 좋을지 감이 잘 안잡혔다. 그러다 최근 Scope function 관련 질문도 받았었겠다, 각각의 Scope function 들에 대해 공부하고 어떤 상황에서 사용하면 좋을지 정리해보았다.  

https://hyeon9mak.github.io/using-kotlin-scope-function-correctly/

**Comment by xrabcde** [comment]
오 !!!! 마지막 플로우차트가 너무 좋네요 코틀린인액션 백날 읽어도 헷갈리던거 한 방에 정리해주는 사진 👍🏻
작성자님은 실제로 코드에서 이 scope function 들을 다 활용하시나요? 🧐
잘 활용하고 싶어도 쓰다보면 항상 쓰던거만 사용하게 되어서.. 쉽지 않네요 ㅋㅎ...

**Reply by Hyeon9mak** [reply]
`apply` 와 `let` 같은 경우 곧장 활용하는데, 나머지는 차이점을 잘 몰랐어서 활용하지 못했네요. 이번 기회에 싹 정리했으니 적극적으로 활용해보려고 합니다 👍

**Comment by tentasys** [comment]
scope function에 대해서 찾아보려고 검색했는데 구글에서 가장 먼저 나오다니 역시 갓갓임당 :+1:
어떤 scope function 사용해야할지 고민되었는데 실마리를 찾은 것 같습니당~~ 항상 배우고잇어용~~~ :bowing_woman:

**Reply by Hyeon9mak** [reply]
오 ㅋㅋㅋ 감사합니다 🙇‍♂️ 
잘 지내시죠?

**Comment by drunkenhw** [comment]
정말 감사합니다 많은 도움이 되었습니다
